src/main/java/minicraft/entity/Arrow.java

```diff
@@ -67,10 +68,17 @@ public void tick() {
 		List<Entity> entitylist = level.getEntitiesInRect(new Rectangle(x, y, 0, 0, Rectangle.CENTER_DIMS));
 		boolean criticalHit = random.nextInt(11) < 9;
 		for (Entity hit : entitylist) {
+			if(hit instanceof ObsidianKnight)this.remove(); //just remove it
 			if (hit instanceof Mob && hit != owner) {
 				Mob mob = (Mob) hit;
 				int extradamage = (hit instanceof Player ? 0 : 3) + (criticalHit ? 0 : 1);
-				mob.hurt(owner, damage + extradamage, dir);
+				if(hit instanceof ObsidianKnight){
+
```

Comment:
Instead of checking twice if the ObsidianKnight is hit, the remove statement from line 71 could be moved inside here.

Reply:
Yes I’ll fix that, it’s pelletsstarPL’s code.